### PR TITLE
Moving Dividend module to Q::F

### PR DESCRIPTION
**Pull request by mm-binary**

### CODE REVIEW

File: MANIFEST [hunk 1]
```diff
@@ -2,6 +2,7 @@
 Changes
```
Comment (jy-binary) [inline MANIFEST]: don't commit MANIFEST file?

Comment (jy-binary) [inline MANIFEST]: sorry ignore that

**Comment by kavehmz** [comment]
@mm-binary  Can you please rewrite the module in Moo instead of Moose.

The change in code will be minimal but increase in speed between these two implementations are noticeable.

https://gist.github.com/kavehmz/e648e2301f02ebe407ff

**Comment by kavehmz** [comment]
@mm-binary Please rewrite the code on Moo instead of Moose.

Change in the code will be minimal but gain in performance will be noticeable.

File: lib/Quant/Framework/Dividend.pm [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+    is         => 'rw',
+    lazy_build => 1,
+);
+
+sub _build_document {
+    my $self = shift;
+
+    my $document = $self->chronicle_reader->get('dividends', $self->symbol);
+
+    if ($self->for_date and $self->for_date->datetime_iso8601 lt $document->{date}) {
```
Comment (basiliscos) [inline lib/Quant/Framework/Dividend.pm]: I feel a bit uneasy, when it is dates are compared as strings. Why not compare `$date->epoch` ?

Comment (mm-binary) [inline lib/Quant/Framework/Dividend.pm]: Done.

Comment (basiliscos) [inline lib/Quant/Framework/Dividend.pm]: :+1:

**Pull request merged**